"Activate" vs "Active" Wording on Theme Customization

illusaen opened the pull request:
Using `theme.isCurrentTheme` to check whether wording on menu bar for "Try and Customize" theme screen should be "Activate" or "Active" instead of always using "Active".

Fixes #17060

## Screenshot
https://user-images.githubusercontent.com/13263478/131718289-7a3fcc9d-6819-45b5-856e-8faa5423bac9.mp4

## Testing
To test:
1. Go to Themes
2. Click menu bar for an **inactive** theme
3. Choose "Try and Customize" on bottom sheet
4. Notice that top right menu bar button should say "Activate"

1. Go to Themes
2. Click menu bar for **active** theme
3. Choose "Try and Customize" on bottom sheet
4. Notice that top right menu bar button should say "Active"

## Regression Notes
1. Potential unintended areas of impact
Using "Activate" on active themes.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested active theme's customization screen as well.

## PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Checking if theme is current theme in order to determine activate/active wording instead of always using active.

Commit:
Updating release notes.

mokagio commented:
@illusaen @enejb I'm code freezing 18.2 today, but this PR hasn't been reviewed yet.

I normally bump PRs that are not ready to the next release, but this being a bug fix I think it's worth keeping for 18.2. I'd like @startuptester opinion on this, too.

I am subscribed to this PR so I'll know if and when it merges and I'll ship a new beta including it when that happens. Or, if we think it can wait till the next release, I'll take care of updating the milestone. 👍

illusaen commented:
> @illusaen @enejb I'm code freezing 18.2 today, but this PR hasn't been reviewed yet.
> 
> I normally bump PRs that are not ready to the next release, but this being a bug fix I think it's worth keeping for 18.2. I'd like @startuptester opinion on this, too.
> 
> I am subscribed to this PR so I'll know if and when it merges and I'll ship a new beta including it when that happens. Or, if we think it can wait till the next release, I'll take care of updating the milestone. 👍

Thanks much!

enejb reviewed (approved):
I tested this and it worked as described. 
And the wording makes sense now. Thanks for the fix!

enejb commented:
Thanks for the fixes @illusaen ! 

During my testing, there are a couple of other things that could be improved but I think those changes can be tacked in different PRs. 

1. When you are customizing the site. The header says "Activate" but pressing it doesn't do anything. 
<img width="400" alt="Screen Shot 2021-09-07 at 12 14 30 PM" src="https://user-images.githubusercontent.com/115071/132398706-3cc2c9e5-2772-48e6-8b05-60496f86d9e2.png">
I am not sure but maybe we can just remove that button completely. 


2. After activating a new theme. You are presented with this pop up. 
<img width="400" alt="Screen Shot 2021-09-07 at 12 16 15 PM" src="https://user-images.githubusercontent.com/115071/132399144-ea786cb6-4337-4cb1-86cf-3361664db0d9.png">

I think we should present the user with something else instead. 
Clicking Manage site doesn't really work as expected. The pop up closes the same thing happens when I click OK. 

3. When the theme is being activated there is no progress indicator. 
4. The "Active" Label is a bit hard to read. 
<img width="400" alt="Screen Shot 2021-09-07 at 12 21 11 PM" src="https://user-images.githubusercontent.com/115071/132399488-9f87da03-5707-439e-ad07-7c49f5db1b68.png">


cc: @iamthomasbishop Would love to know how we should tackle some of the issues that I found here. 
They could definitely be their own tickets here.

mokagio commented:
I'm planning to ship a new beta today to start testing https://github.com/wordpress-mobile/WordPress-iOS/pull/17130.

I'll be "taking over" this PR to get it in the beta. Whenever we can batch fixes into betas we try to do it, as deploying a beta involves quite a bit of extra communications downstream.

mokagio committed:
Merge remote-tracking branch 'origin/release/18.2' into theme-active-vs-activate

GitHub couldn't automatically resolve a conflict on the
`RELEASE-NOTES.txt` file, but my local Git auto-merged it without
trouble...

```
Auto-merging RELEASE-NOTES.txt
Merge made by the 'recursive' strategy.
```

mokagio committed:
Remove redundant entry from `RELEASE-NOTES.txt`

In d9112e0ef37a543e8980fea56aaee82e84981b0b, I boasted about how my
local Git was able to automatically resolve merge conflicts on
`RELEASE-NOTES.txt`. Too bad it did it in a way that was not appropriate 🤦‍♂️

I should have checked that before pushing the commit... This addresses
that little issue.

mokagio commented on RELEASE-NOTES.txt:
```diff
@@ -6,6 +6,7 @@
 -----
 * [internal] Fixed an issue where source and platform tags were not added to a Zendesk ticket if the account has no blogs. [#17084]
 * [*] Set the post formats to have 'Standard' first and then alphabetized the remaining items. [#17074]
+* [*] Fixed wording of theme customization screen's menu bar by using "Activate" on inactive themes. [#17060]
```
@illusaen note that this entry won't make it into the release notes in the App Store, because those have already been submitted for translation by #17131.

Given the content, I think that's okay, but let me know if you think otherwise.

mokagio commented:
@illusaen   this has been bundled as part of [18.2 beta 1 (18.2.0.1)](https://github.com/wordpress-mobile/WordPress-iOS/pull/17136).

Thanks for your work 🙌
